src/linear_algebra/free_module.lean

```diff
@@ -131,17 +133,44 @@ instance pi {ι : Type*} [fintype ι] {M : ι → Type*} [Π (i : ι), add_comm_
 [Π (i : ι), module R (M i)] [Π (i : ι), module.free R (M i)] : module.free R (Π i, M i) :=
 of_basis $ pi.basis $ λ i, choose_basis R (M i)
 
+instance matrix {ι₁ : Type*} [fintype ι₁] {ι₂ : Type*} [fintype ι₂] :
+  module.free R (matrix ι₁ ι₂ R) :=
+of_basis $ pi.basis $ λ i, pi.basis_fun _ _
```

Comment:
Can you make a def for `pi.basis $ λ i, pi.basis_fun _ _` as `matrix.basis`?

Reply:
I've added `matrix.std_basis` to `linear_algebra.std_basis`. I am not sure it is the best file, but I don't see anything better. Note that `linear_algebra.matrix.basis` concerns basis of the modules, not of the matrix space itself.